### DISCUSSION TITLE
Improved latency via USB 2.0 on Raspberry Pi 4 -> Bug in Latency Display or Hidden Feature?

### DISCUSSION THREAD

**Discussion by uklug** [jamulussoftware/jamulus#2203] [Bug Investigation]
Dear Jamulus-Team, thank you so much for this great app! Jamulus helped our band to continue jamming and practicing during this strange time. 
During the last months I tested many configurations for the jamulus client under windows and linux - the latter on pcs and also a raspberry pi 4 operated with the latest 64bit-RaspiOS bullseye. Latency-Performance of both systems is more or less the same with the latest releases, if the same audio-interface is used. However, by incidence I figured out that an older USB-Interface driven at the USB 2.0 host of the raspi 4 was suddenly 3ms better than on the pc (buffer settings without automation: local: 3 / server 3, ping: 10ms, total jamulus delay on pc 22ms, total jamulus delay on raspi 4: 19ms!). This "low-latency-feature" is reproducable, but only combined with the USB 2.0 host. The same old interface plugged to the usb3 host will again only operate at 22 ms. And now the very strange thing: When I remove the old usb2-interface and plug a USB3 audio interface into the same USB2-slot, this new interface will now also operate faster at 19ms (well, at least the display says so). However, this improved latency will only persist as long as the raspi is on: after reboot the improved latency is gone.... until I restart the game with the older interface again. Is this a bug in the latency display or is there some latency treasure hidden in the USB-drivers? I hope there's somebody believing my story.

**Comment by pljones** [comment]
Jamulus has no visibility of this.  It talks to JACK.  So you'd need to look to the USB drivers.
